sdk/internal/temporal/resource.go

```diff
@@ -121,3 +147,7 @@ func (er *Resource[TResource, TState]) Expire() {
 	// Reset the expiration as if we never got this resource to begin with
 	er.expiration = time.Time{}
 }
+
+func (er *Resource[TResource, TState]) expiringSoon(TResource, TState) bool {
+	return er.expiration.Add(-5 * time.Minute).Before(time.Now())
```

Comment:
Previously, we were using `now` which was computed at the beginning of `Get()`. Now, we're calculating a new `now` (locally) which is no longer constant in the `for` loop. Is this intentional, or should `expiringSoon` take a `time.Time` param which is the loop's `now`?

Reply:
It's intentional in that ShouldRefresh gets only the resource and the state and adding `now time.Time` there is too weird for me. I suppose recomputing now is harmful in that it arguably wastes some number of cycles, but do you see any possibility of incorrect behavior? I suppose it could trigger a refresh some nanoseconds earlier than expected, but I don't see harm in that because Get observes backoff (although it does shrink the backoff window by setting `er.lastAttempt` to the clock time before acquiring the lock and refreshing the resource)

Reply:
Yeah I agree adding a `now time.Time` param would be weird.

Mostly I was worried that the `for` loop could somehow become live-locked due to drifting values of `now` and `local now` (I think there's a Space Balls joke in here somewhere). But since it doesn't sleep then I think that can't happen.

Reply:
It would be good to add a comment though that `time.Now()` is intentional.